lfs/scanner.go

```diff
@@ -192,7 +195,8 @@ func (m *indexFileMap) Set(sha string, index *indexFile) {
 // ScanIndex returns a slice of WrappedPointer objects for all
 // Git LFS pointers it finds in the index.
 // Reports unique oids once only, not multiple times if >1 file uses the same content
-func ScanIndex() ([]*WrappedPointer, error) {
+// isFirstCommit must be true if this repo doesn't have any commits yet
+func ScanIndex(isFirstCommit bool) ([]*WrappedPointer, error) {
```

Comment:
Nice. What do you think about changing this to parameter to instead be the first ref to begin with? I think that might make the behavior of the function more obvious to the caller (instead of having to look and see what `true`/`false` means).

The `nullTreeish` constant makes this a little tricky to deal with, so maybe it could be called `InitialCommit` or something similar?

`ScanIndex("HEAD")`, `ScanIndex(InitialCommit)`.